core/store/ChainStore/ChainStore.go

```diff
@@ -759,10 +749,50 @@ func (bd *ChainStore) persist(b *Block) error {
 				}
 			}
 		}
+		
+		for index := 0; index < len(b.Transactions[i].Outputs); index++ {
+			output := b.Transactions[i].Outputs[index]
+			programHash := output.ProgramHash
+			assetId := output.AssetID
+			if value, ok := accounts[programHash]; ok {
+				value.Balances[assetId] += output.Value
```

Comment:
it is unnecessary to check assetid. if account exist, balances has been instantiated.